operator/pkg/helmreconciler/apply.go

```diff
@@ -193,7 +203,11 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured) error {
 			if err := applyOverlay(receiver, obj); err != nil {
 				return err
 			}
-			return h.client.Update(context.TODO(), receiver)
+			if err := h.client.Update(context.TODO(), receiver); err != nil {
+				return fmt.Errorf("failed to update %q: %w", objectStr, err)
```

Comment:
I am think this will break `RetryOnConflict` because we lose the error context? Or does `%w` magically handle that?

Reply:
Yes. It breaks. Fixed it.